pkg/openstackbaremetalset/baremetalhost.go

```diff
@@ -119,75 +188,6 @@ func BaremetalHostProvision(
 		preProvNetworkData = instance.Spec.BaremetalHosts[hostName].PreprovisioningNetworkDataName
 	}
 
-	if networkDataSecret == nil && preProvNetworkData == "" {
```

Comment:
Do we still have this logic somehow represented in the reworked version?  If I understand correctly, we don't want to create a `networkDataSecret` if a `preProvNetworkData` secret has been set for the host [1].  @rabi can you keep me honest here? :)

[1] https://github.com/openstack-k8s-operators/openstack-baremetal-operator/blob/615a9be17dc678f42c7aae663a323a2a304e3de5/pkg/openstackbaremetalset/baremetalhost.go#L186-L189

Reply:
I've dropped that because nobody could really tell me why. If there is a good reason for it, the conditional can get back in. Along with error to be raised in case it fails obviously.

Reply:
There still seems to be some discussion to be had about the desire behavior. So for now the behavior will remain the same. I've restored conditional on `PreprovisioningNetworkDataName`. Now the PR is purely housekeeping effort.

When there is a decision on what should take primacy, it's just a question of minor adjustment.